src/main/java/picard/sam/SAMSplitter.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.sam;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.Log;
+import htsjdk.samtools.util.ProgressLogger;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import picard.PicardException;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.SamOrBam;
+
+import java.io.File;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * <p/>
+ * Splits the input queryname sorted or querygrouped SAM/BAM file and writes it into
+ * multiple BAM files of approximately equal size. This will retain the sort order
+ * within each output BAM and if the BAMs are concatenated in order (output files are named
+ * numerically) the order of the reads will match the original BAM. As this tool is
+ * intended for unmapped input files, all secondary and supplementary reads will be removed.
+ */
+@CommandLineProgramProperties(
+        summary = SAMSplitter.USAGE_SUMMARY + SAMSplitter.USAGE_DETAILS,
+        oneLineSummary = SAMSplitter.USAGE_SUMMARY,
+        programGroup = SamOrBam.class)
+@DocumentedFeature
+public class SAMSplitter extends CommandLineProgram {
```

Comment:
1. `SAM` -> `Sam` everywhere.  See how other tools are named please.
2. Can we rename this to `SplitBamBySize`?  This is more in line with the names of other tools.  Avoid verbs, use nouns.